### ISSUE TITLE
usage logging

### ISSUE THREAD

**Issue by bitno** [coneda/kor#127]
Dear @moritzschepp 
Is KOR logging the usage/access to its public interface? Meaning, would I find a log with all clicks on a certain entity, eg. https://conedakor8.ethz.ch/blaze#/entities/764?
Many thanks and kind regards,
Leo

**Comment by moritzschepp** [comment]
Hey @bitno, yes it does, that's the file ´log/production.log´. Every request is logged there and depending on how you configured the log level in ´config/environments/production.rb´, there is more or less information about every request. The specific url you are interested in is pointing to a page with a lot of javascript going on so you should expect many (ajax) requests for one user looking at the page.

**Comment by bitno** [comment]
Dear @moritzschepp 
Ok, many thanks. Is there a way how one could easily filter the production log? 
Or let me rephrase it, how much effort would it be, to embed visitor statistics as a single measure onto each entity? Something like: www.kor.com/MonaLisa - page visited 3200 times
Many thanks

**Comment by moritzschepp** [comment]
A simple way is to use `grep` as in:

    cat log/production.log | grep "Started GET \"/entities/335037"

and then this gives you the count of lines, words and characters:

    cat log/production.log | grep "Started GET \"/entities/335037" | wc

This would include calls to the API that were made by other clients than your KOR frontend though. If you would like to have more specifics, a tool like piwik might help: Then the browser communicates the tracking info and it knows exactly what pages have been visited.

**Comment by bitno** [comment]
Much appreciated, thank you!!!!!! Going to test this now

**Comment by moritzschepp** [comment]
BTW, the production.log file should be rotated by logrotate if that is the case, then there might be a bunch of those files in the log directory, some of them gzipped. If you want to go through all of them, you can use

    zcat -f log/production* | grep ...

instead of the normal cat command.

**Comment by bitno** [comment]
in our case not yet, which is in this particular case is a good thing ;D
